salt/grains/core.py

```diff
@@ -399,6 +399,73 @@ def _sunos_cpudata():
     return grains
 
 
+def _aix_cpudata():
+    '''
+    Return CPU information for AIX systems
+    '''
+    # Provides:
+    #   cpuarch
+    #   num_cpus
+    #   cpu_model
+    #   cpu_flags
+    grains = {}
+    cmd = '/usr/sbin/prtconf'
+    data = ""
+    if salt.utils.path.which(cmd):  # Also verifies that cmd is executable
+        data += __salt__['cmd.run']('{0}'.format(cmd))
+        data += '\n'
+
+    cpuarch_regexes = [
+        re.compile(r) for r in [
+            r'(?im)^\s*Processor\s+Type:\s+(\S+)',  # prtconf
+        ]
+    ]
+
+    cpuflags_regexes = [
+        re.compile(r) for r in [
+            r'(?im)^\s*Processor\s+Version:\s+(\S+)',  # prtconf
+        ]
+    ]
+
+    cpumodel_regexes = [
+        re.compile(r) for r in [
+            r'(?im)^\s*Processor\s+Implementation\s+Mode:\s+(.*)',  # prtconf
+        ]
+    ]
+
+    numcpus_regexes = [
+        re.compile(r) for r in [
+            r'(?im)^\s*Number\s+Of\s+Processors:\s+(\S+)',  # prtconf
+        ]
+    ]
+
+    for regex in cpuarch_regexes:
+        res = regex.search(data)
+        if res and len(res.groups()) >= 1:
+            grains['cpuarch'] = res.group(1).strip().replace("'", "")
+            break
+
+    for regex in cpuflags_regexes:
+        res = regex.search(data)
+        if res and len(res.groups()) >= 1:
+            grains['cpu_flags'] = res.group(1).strip().replace("'", "")
+            break
+
+    for regex in cpumodel_regexes:
+        res = regex.search(data)
+        if res and len(res.groups()) >= 1:
+            grains['cpu_model'] = res.group(1).strip().replace("'", "")
+            break
+
+    for regex in numcpus_regexes:
+        res = regex.search(data)
+        if res and len(res.groups()) >= 1:
+            grains['num_cpus'] = res.group(1).strip().replace("'", "")
+            break
```

Comment:
OK, how about compacting all this into just couple of lines like so?
```python
for dest, regstring in (('cpuarch', r'(?im)^\s*Processor\s+Type:\s+(\S+)'),
                        ('cpu_flags', r'(?im)^\s*Processor\s+Version:\s+(\S+)'),
                        ('cpu_model', r'(?im)^\s*Processor\s+Implementation\s+Mode:\s+(.*)'),
                        ('num_cpus', r'(?im)^\s*Number\s+Of\s+Processors:\s+(\S+)')):
    for regex in [re.compile(r) for r in [regstring]]:
        res = regex.search(data)
        if res and len(res.grouns()) >= 1:
            grains[dest] = res.group(1).strip().replace("'", '')
```
Unit tests are, of course, welcome. :wink:

Reply:
Following style used for Solaris in the file.